### ISSUE TITLE
This is a very interesting job, is it possible to make the code public.

### ISSUE THREAD

**Issue by guangmingjian** [cedricvincentcuaz/TFGW#1]
I'm curious and unsure about the specific implementation of the thesis part. For example: how to connect the POT library with pytorch without affecting backpropagation.

**Comment by cedricvincentcuaz** [comment]
Hello @guangmingjian 

I updated the readme to mention that we are currently working on a factorised implementation with a pipeline compatible with the user-defined GNN pre-processing architecture, which will be released in a few days. You can find an exemple with GIN pre-processing, as set by default in the paper, in the supplementary material shared during the reviewing process. Link: [vanilla implementation](https://openreview.net/attachment?id=seYcx6CqPe&name=supplementary_material). The code to reproduce experiments is there too, except those with GAT architectures.

We are also working on speeding up FGW solvers implemented in POT. This will allow a 20% to 50% gain in speed over graphs from 10 to 1000 nodes, compared to POT 0.8.2. These new solvers will be released soon in POT 0.8.3 . Basically, the FGW2 function of POT manually sets the gradients to allow back-propagation through the FGW loss.

Best regards :)
